jolt-core/src/jolt/vm/mod.rs

```diff
@@ -119,7 +126,15 @@ pub trait Jolt<F: PrimeField, G: CurveGroup<ScalarField = F>, const C: usize, co
       &mut random_tape,
     );
 
-    todo!("Lasso lookups to enforce timestamp validity")
+    let timestamp_validity_lookups: Vec<SLTUInstruction> = read_timestamps
+      .iter()
+      .enumerate()
+      .map(|(i, &ts)| SLTUInstruction(ts, i as u64 + 1))
```

Comment:
It should also work with `ceil(i / 7)` here, which is when we increment the counter only once per CPU step.